pandas/core/sparse/array.py

```diff
@@ -120,298 +136,458 @@ def _wrap_result(name, data, sparse_index, fill_value, dtype=None):
         # fill_value may be np.bool_
         fill_value = bool(fill_value)
-    return SparseArray(data, sparse_index=sparse_index,
-                       fill_value=fill_value, dtype=dtype)
+    return SparseArray(data,
+                       sparse_index=sparse_index,
+                       fill_value=fill_value,
+                       dtype=dtype)
 
 
-class SparseArray(PandasObject, np.ndarray):
-    """Data structure for labeled, sparse floating point 1-D data
+class SparseArray(PandasObject, ExtensionArray, ExtensionOpsMixin):
+    """
+    An ExtensionArray for storing sparse data.
+
+    .. versionchanged:: 0.24.0
+
+       Implements the ExtensionArray interface.
 
     Parameters
     ----------
-    data : {array-like (1-D), Series, SparseSeries, dict}
-    kind : {'block', 'integer'}
-    fill_value : float
-        Code for missing value. Defaults depends on dtype.
-        0 for int dtype, False for bool dtype, and NaN for other dtypes
-    sparse_index : {BlockIndex, IntIndex}, optional
-        Only if you have one. Mainly used internally
-
-    Notes
-    -----
-    SparseArray objects are immutable via the typical Python means. If you
-    must change values, convert to dense, make your changes, then convert back
-    to sparse
+    data : array-like
+        A dense array of values to store in the SparseArray. This may contain
+        `fill_value`.
+    sparse_index : SparseIndex, optional
+    index : Index
+    fill_value : scalar, optional
+        Elements in `data` that are `fill_value` are not stored in the
+        SparseArray. For memory savings, this should be the most common value
+        in `data`. By default, `fill_value` depends on the dtype of `data`:
+
+        =========== ==========
+        data.dtype  na_value
+        =========== ==========
+        float       ``np.nan``
+        int         ``0``
+        bool        False
+        datetime64  ``pd.NaT``
+        timedelta64 ``pd.NaT``
+        =========== ==========
+
+        When ``data`` is already a ``SparseArray``, ``data.fill_value``
+        is used unless specified, regardless of `data.dtype``.
+
+    kind : {'integer', 'block'}, default 'integer'
+        The type of storage for sparse locations.
+
+        * 'block': Stores a `block` and `block_length` for each
+          contiguous *span* of sparse values. This is best when
+          sparse data tends to be clumped together, with large
+          regsions of ``fill-value`` values between sparse values.
+        * 'integer': uses an integer to store the location of
+          each sparse value.
+
+    dtype : np.dtype or SparseDtype, optional
+        The dtype to use for the SparseArray. For numpy dtypes, this
+        determines the dtype of ``self.sp_values``. For SparseDtype,
+        this determines ``self.sp_values`` and ``self.fill_value``.
+    copy : bool, default False
+        Whether to explicitly copy the incoming `data` array.
     """
+
     __array_priority__ = 15
```

Comment:
can remove this

Reply:
This is still required for older numpy versions not supporting array_ufunc.